docarray/index/backends/redis.py

```diff
@@ -0,0 +1,606 @@
+        index_name: Optional[str] = None
+        username: Optional[str] = None
+        password: Optional[str] = None
+        text_scorer: str = field(default='BM25')
+        default_column_config: Dict[Type, Dict[str, Any]] = field(
+            default_factory=lambda: defaultdict(
+                dict,
+                {
+                    VectorField: {
+                        'algorithm': 'FLAT',
+                        'distance': 'COSINE',
+                        'ef_construction': None,
+                        'm': None,
+                        'ef_runtime': None,
+                        'initial_cap': None,
+                    },
+                },
+            )
+        )
+
+        def __post_init__(self):
+            self.text_scorer = self.text_scorer.upper()
+
+            if self.text_scorer not in VALID_TEXT_SCORERS:
+                raise ValueError(
+                    f"Invalid text scorer '{self.text_scorer}' provided. "
+                    f"Must be one of: {', '.join(VALID_TEXT_SCORERS)}"
+                )
+
+    @dataclass
+    class RuntimeConfig(BaseDocIndex.RuntimeConfig):
+        """Dataclass that contains all "dynamic" configurations of RedisDocumentIndex.
+
+        :param batch_size: Batch size during indexing.
+        """
+
+        batch_size: int = 100
+
+    def python_type_to_db_type(self, python_type: Type) -> Any:
+        """
+        Map python types to corresponding Redis types.
+
+        :param python_type: Python type.
+        :return: Corresponding Redis type.
+        """
+        type_map = {
+            int: NumericField,
+            float: NumericField,
+            str: TextField,
+            bytes: TextField,
+            np.ndarray: VectorField,
+            list: VectorField,
+            AbstractTensor: VectorField,
+        }
+
+        for py_type, redis_type in type_map.items():
+            if issubclass(python_type, py_type):
+                return redis_type
+        raise ValueError(f'Unsupported column type for {type(self)}: {python_type}')
+
+    @staticmethod
+    def _generate_items(
+        column_to_data: Dict[str, Generator[Any, None, None]],
+        batch_size: int,
+    ) -> Iterator[List[Dict[str, Any]]]:
+        """
+        Given a dictionary of generators, yield a list of dictionaries where each
+        item consists of a key and a single item from the corresponding generator.
+
+        :param column_to_data: A dictionary where each key is a column and each value
+            is a generator.
+        :param batch_size: Size of batch to generate each time.
+
+        :yield: A list of dictionaries where each item consists of a column name and
+            an item from the corresponding generator. Yields until all generators
+            are exhausted.
+        """
+        keys = list(column_to_data.keys())
+        iterators = [iter(column_to_data[key]) for key in keys]
+        batch: List[Dict[str, Any]] = []
+
+        while True:
```

Comment:
is this `while True` needed?

Reply:
yes because I don't know the size in advance

Reply:
how do u break out of the loop?

Reply:
Review this because I feel is not needed. it seems that u will infintely loop over the same keys and iterator

Reply:
One item is created on each iteration, stops when the iterator is exhausted, i.e. `item` is None (in this case I'm checking the iterator for ID because others might have None values)

```python
if key == 'id' and not item:
    if batch:
        yield batch
  return
```

What could be dangerous here?